base/reshapedarray.jl

```diff
@@ -291,3 +291,7 @@ substrides(strds::NTuple{N,Int}, I::Tuple{ReshapedUnitRange, Vararg{Any}}) where
     (size_to_strides(strds[1], size(I[1])...)..., substrides(tail(strds), tail(I))...)
 unsafe_convert(::Type{Ptr{T}}, V::SubArray{T,N,P,<:Tuple{Vararg{Union{RangeIndex,ReshapedUnitRange}}}}) where {T,N,P} =
     unsafe_convert(Ptr{T}, V.parent) + (first_index(V)-1)*sizeof(T)
+
+## ReshapedArray eltype conversion
+
+map(::Type{T}, R::ReshapedArray) where {T} = ReshapedArray(map(T, parent(R)), R.dims, R.mi)
```

Comment:
Why only for types? It seems confusing for `map(Float64, x)` to produce something different from `map(x -> Float64(x), x)`.

Reply:
There are special `map` methods defined for ranges that dispatch on types. Currently

```julia
julia> map(Int, 3:4)
3:4

julia> map(x -> Int(x), 3:4)
2-element Vector{Int64}:
 3
 4
```

It'll certainly be ideal if this may be addressed in general. Any suggestions?

Reply:
Ah, yeah. In that case it is unavoidable unfortunately, since we can't really tell in general whether a function will preserve the range structure of an `AbstractRange`, but we know that numeric type constructors will. In the case here, I don't really see a reason to restrict this though.